aggregator/src/config.rs

```diff
@@ -147,6 +147,38 @@ pub struct JobDriverConfig {
     pub maximum_attempts_before_failure: usize,
 }
 
+// TODO(#2252): This custom deserializer is for backwards-compatibility with Janus 0.6.3 and below.
+// Remove it for the next major version of Janus.
+impl<'de> Deserialize<'de> for JobDriverConfig {
```

Comment:
This is fine, but I wonder if either [`serde(rename)`](https://serde.rs/field-attrs.html#rename) or [`serde(alias)`](https://serde.rs/field-attrs.html#alias) could do the trick with less code?

Reply:
`#[serde(alias = "max_job_discovery_delay_secs")]` works great, but we then lose the ability to `warn!` when falling back to a deprecated config. IMO the warning is important as a heads up for the next minor release (or in our case, heads up that the environment is missing the updated configuration).